src/api/UserApi.ts

```diff
@@ -1,14 +1,23 @@
-import {Controller, Post, Request, Route, Tags} from 'tsoa'
+import {Controller, Get, Post, Request, Route, Security, Tags} from 'tsoa'
 import Response from "common/Response";
 import {Request as ExpressRequest} from 'express'
 import UserService from "service/UserService";
 import {User} from "models/User";
-
+import { AuthRequest } from 'common/AuthRequest';
 
 @Route()
 @Tags('User Api')
 export class UserApi extends Controller {
 
+    @Security('GOOGLE_TOKEN')
+    @Get('/users/me')
```

Comment:
Jak dla mnie to powinien być endpoint w stylu `/profile` lub `/me` bez przedrostka `/users`. Normalnie jest endpoint `/users/:id`, więc zostawiłbym taką możliwość na przyszłość i zmienił ten endpoint. @Mitelak co uważasz?

Reply:
Takie rozwiązanie nie powoduje, że nie możesz w przyszłości dorobić `/users/:id`, a wręcz przeciwnie jest dokładnie implementacją specyficznego przypadku tego endpointa. 

Jeżeli chcesz odpytać endpoint `/users/:id` o swoje dane to zazwyczaj jest problem, że nie znasz swojego ID, bo nie masz jeszcze danych z endpointa. W takim przypadku wydzielony jest case gdzie pod `/:id` podkładasz `/me` i wiadomo, że chodzi o usera, który robi request, a zwrotka/zachowanie jest dokładnie takie samo jak dla generycznego endpointa `/users/:id`. 

Zrobiłem tak, bo widziałem już takie implementacje i IMO jest to spoko podojeście, ale czy to będzie `/profile`, czy `/users/me` to mi całkowicie obojętne 🤷🏻 

(no może nie całkowicie obojętne, bo nie chce mi się zmieniać, ale jak trzeba to trzeba, up to you 😂 )

Reply:
No dla mnie to nie jest poprawne, a posłużę się tutaj przykładem. Jak mówisz, że `users/me` to specyficzny przykład `users/:id` to dlaczego `me` nie będzie się zgadzało ze standardem reszty idików? Do tego jak będziemy chcieli kiedyś stworzyć endpoint `users/:id` będziemy musieli pamiętać o tym, żeby było to łapane znacznie później niż endpoint `users/me`, bo inaczej `me` zostanie zinterpretowany jako jeden z `id` tylko, że dla tego przypadku nie znajdzie takiego użytkownika, bo nie będzie tłumaczenia wartości `me` na prawdziwe `id` danego użytkownika. Widzę, że Kopek dał okejkę dla mojej pierwszej wiadomości, więc jednak fajnie jak to zmienisz ;)